### PR TITLE
feat: add tab context menu for code-editor

### PR DESCRIPTION
**Pull request by iNorthIsle**
added some context menus of tab in code-editor section:

- [x] close
- [x] close others
- [x] close all
- [x] copy path
- [x] open containing folder

Note: the copy path format of a file is different between online editor and desktop editor:
- online editor: `projectName/**/filename`
- desktop editor: `/Users/username/dir/**/projectName/**/filename`

will add the menu of the copy relative path in another pr.

**web-editor screenshots:**
> web editor should not have the "Open Containing Folder"

| file tree      |  tab |
| ----------- | ----------- |
image|image

**desktop-editor screenshots:**
| file tree      |  tab |
| ----------- | ----------- |
image|image

### CODE REVIEW

File: packages/code-editor/src/CodeEditorCollection.js [hunk 1]
```diff
@@ -23,6 +23,29 @@ export default class CodeEditorCollection extends PureComponent {
...
+        text: 'Close All',
+        onClick: this.closeAll
+      },
+      null,
+      {
+        text: 'Copy Path',
+        onClick: this.copyPath
+      },
+      ...platform.isDesktop ? [{
+        text: 'Reveal In Finder',
```
Comment (qftgtr) [inline packages/code-editor/src/CodeEditorCollection.js:45]: Open Containing Folder
see context menu for the file tree

Comment (iNorthIsle) [reply packages/code-editor/src/CodeEditorCollection.js]: updated: https://github.com/ObsidianLabs/electron-components/pull/56/files#diff-06ff2e4ab637b04f57df25a54798f2a4793cc05670bdf87abe1c3ac9e196bb29R45

File: packages/ui-components/src/Tabs/TabHeader.js [hunk 2]
```diff
@@ -51,25 +53,47 @@ class TabHeaderItem extends PureComponent {
...
+                onClick: e => e.preventDefault()
+              }}
+            >
+              <div className='nav-item-text' onMouseDown={e => {
+                console.log(e)
+                e.button === 0 && onSelectTab(tab)
+
+              }}
+                onMouseUp={e => {
+                  console.log(e)
```
Comment (qftgtr) [inline packages/ui-components/src/Tabs/TabHeader.js:79]: delete console.log

File: packages/ui-components/src/Tabs/TabHeader.js [hunk 3]
```diff
@@ -51,25 +53,47 @@ class TabHeaderItem extends PureComponent {
...
-                {tabText}
+            <ContextMenuTrigger
+              id={`${tab.key}/tab-item`}
+              ref={ref => { this.clickableRef = ref }}
+              attributes={{
+                onClick: e => e.preventDefault()
+              }}
+            >
+              <div className='nav-item-text' onMouseDown={e => {
+                console.log(e)
```
Comment (qftgtr) [inline packages/ui-components/src/Tabs/TabHeader.js:74]: delete console.log

File: packages/platform/rollup.config.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
+import babel from '@rollup/plugin-babel'
```
Comment (qftgtr) [inline packages/platform/rollup.config.js:1]: do not use rollup in platform

File: packages/code-editor/src/CodeEditorCollection.js [hunk 5]
```diff
@@ -166,13 +208,15 @@ export default class CodeEditorCollection extends PureComponent {
       <div className='d-flex w-100 h-100 overflow-hidden bg2'>
         <Tabs
           ref={this.tabs}
-          size='sm'
+          size='xl'
+          asd='1'
```
Comment (qftgtr) [inline packages/code-editor/src/CodeEditorCollection.js:212]: remove

File: packages/platform/package.json [hunk 6]
```diff
@@ -4,10 +4,40 @@
...
+  "main": "main/index.js",
+  "browser": "dist/index.es.js",
   "files": [
-    "src"
+    "dist",
+    "main"
   ],
-  "scripts": {},
-  "dependencies": {}
+  "scripts": {
```
Comment (qftgtr) [inline packages/platform/package.json:14]: remove scripts and dependencies; keep platform minimal

File: packages/ui-components/src/Tabs/TabContextMenu.js [hunk 7]
```diff
@@ -0,0 +1,52 @@
+import React, { PureComponent } from 'react'
+import { ContextMenu, MenuItem } from 'react-contextmenu'
+
+ContextMenu.defaultProps.className = 'dropdown-menu dropdown-menu-sm show'
+MenuItem.defaultProps.className = 'dropdown-item'
```
Comment (qftgtr) [inline packages/ui-components/src/Tabs/TabContextMenu.js:5]: do not use defaultProps

**Pull request merged**